src/useImageDimensions.ts

```diff
@@ -0,0 +1,36 @@
+import {useEffect, useState} from 'react'
+import {Image, ImageRequireSource} from 'react-native'
+
+export interface URISource {
```

Comment:
You can import `ImageURISource` from `react-native` instead of specifying your own

Reply:
There is a lot of redundant properties in ImageURISource, that are not using in the hook.

Reply:
Those properties are optional though? I think that there is value in being compatible with upstream.

e.g. if they add a third way to load images in the future, it will automatically work with that as well. Or if another platform (e.g. macOS, Windows, Web) uses another format, and overrides the types, it will also work.

Reply:
I prefer to use a small interface. If there will appear any other functionality in different platforms so we will be able to extend the functionality of this hook too.

Reply:
I would still prefer to use the built-in interface instead of inventing our own here. That way the documentation also will show something that users are potentially already familiar with, and since we are passing the props to e.g. `Image.resolveAssetSource` some of those properties could potentially affect what is returned.

Reply:
Could you please provide an example on which properties will affect the result?

Reply:
My point was that we should try and be proactive in our coding. It's not really about wether or not there _is_ an attribute right now that would affect that, but if one is added (or already exists, I haven't checked) we would automatically support that.

Reply:
I am proactive in my code. See the answer in the thread below.